File: labs/solidworks/swex/add-in/Overview.DocHandler.vb
'...
Private m_DocHandler As IDocumentsHandler(Of DocumentHandler)
'...
    '...
    m_DocHandler = CreateDocumentsHandler()
    AddHandler m_DocHandler.HandlerCreated, AddressOf OnDocHandlerCreated
    '...
Private Sub OnDocHandlerCreated(ByVal doc As DocumentHandler)
    AddHandler doc.Rebuild, AddressOf OnDocRebuild
    AddHandler doc.Save, AddressOf OnDocSave
End Sub

Private Function OnDocRebuild(ByVal docHandler As DocumentHandler, ByVal state As RebuildState_e) As Boolean
    'TODO: handle rebuild
    Return True
End Function

Private Function OnDocSave(ByVal docHandler As DocumentHandler, ByVal fileName As String, ByVal state As SaveState_e) As Boolean
    'TODO: handle saving
    Return True
End Function

File: labs/solidworks/swex/add-in/Overview.DocHandler.cs
//...
private IDocumentsHandler<DocumentHandler> m_DocHandler;
//...
    //...
    m_DocHandler = CreateDocumentsHandler();
    m_DocHandler.HandlerCreated += OnDocHandlerCreated;
    //...
private void OnDocHandlerCreated(DocumentHandler doc)
{
    doc.Rebuild += OnDocRebuild;
    doc.Save += OnDocSave;

}

private bool OnDocRebuild(DocumentHandler docHandler, RebuildState_e state)
{
    //TODO: handle rebuild
    return true;
}

private bool OnDocSave(DocumentHandler docHandler, string fileName, SaveState_e state)
{
    //TODO: handle saving
    return true;
}